### PR TITLE
removes unused wallet code for syncing transactions

### PR DESCRIPTION
**Pull request by hughy**
## Summary

removes syncTransaction from wallet and account.

removes SyncTransactionParams.

removes bulkUpdateDecryptedNotes and processTransactionSpends which were only called from syncTransaction.

removes updateDecryptedNote and replaces remaining uses with calls to saveDecryptedNote.

removes updateTransaction and replaces remaining uses with calls to saveTransaction.

removes unnecessary join with transactions in getUnspentNotes; uses decryptedNote.sequence for block confirmations.

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish/src/wallet/account.ts [hunk 1]
```diff
@@ -548,7 +395,8 @@ export class Account {
             'nullifierToNote mappings must have a corresponding decryptedNote',
           )
 
-          await this.updateDecryptedNote(
+          await this.walletDb.saveDecryptedNote(
```
Comment (danield9tqh) [inline ironfish/src/wallet/account.ts:398]: So I understand this correctly. The only real logic change here is that we are no longer attempting to delete the nullifier for these notes. And that is because spends will always have nullifiers so no reason to ever delete them?

File: ironfish/src/wallet/wallet.ts [hunk 2]
```diff
@@ -957,7 +903,7 @@ export class Wallet {
 
     if (noteMapValue) {
       this.logger.debug(`Unspent note has index ${String(noteMapValue.index)}`)
-      await sender.updateDecryptedNote(unspentNote.hash, {
+      await this.walletDb.saveDecryptedNote(sender, unspentNote.hash, {
```
Comment (danield9tqh) [inline ironfish/src/wallet/wallet.ts:906]: Same for this change.. We already do an early return if there are no nullifiers so no need to check deletion of nullifier here either?

File: ironfish/src/wallet/wallet.ts [hunk 3]
```diff
@@ -676,20 +631,11 @@ export class Wallet {
...
-
-        Assert.isNotUndefined(
-          transaction,
-          `Transaction '${decryptedNote.transactionHash.toString(
-            'hex',
-          )}' missing for account '${account.id}'`,
-        )
-
-        if (!transaction.sequence) {
+        if (!decryptedNote.sequence) {
```
Comment (danield9tqh) [inline ironfish/src/wallet/wallet.ts:634]: I know this was done in another PR but curious why we moved blockHash/sequence to the note since it seems like a denormalization if the note also has the transactionHash

**Review by danield9tqh — APPROVED** [review]

File: ironfish/src/wallet/account.ts (hunk 1 above)
Comment (hughy) [reply ironfish/src/wallet/account.ts:398]: Yeah exactly, the spends will still have nullifiers, and those would be removed only if the transaction is disconnected from the chain

File: ironfish/src/wallet/wallet.ts (hunk 3 above)
Comment (hughy) [reply ironfish/src/wallet/wallet.ts:634]: We added blockHash and sequence to the note for performance -- avoiding looking up the transaction to get the sequence for a note

File: ironfish/src/wallet/wallet.ts (hunk 2 above)
Comment (hughy) [reply ironfish/src/wallet/wallet.ts:906]: Yeah we know that the note has a nullifier already, so we wouldn't delete the index entry in this case
